res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -0,0 +1,974 @@
+    engine.softTakeover("[Master]", "gain", true);
+    engine.softTakeover("[Master]", "headMix", true);
+    engine.softTakeover("[Master]", "headGain", true);
+
+    // Dirty hack to set initial values in the packet parser
+    var data = TraktorS2MK3.toBytes("01 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 00 00 00 00");
+    TraktorS2MK3.incomingData(data);
+};
+
+TraktorS2MK3.registerInputJog = function (message, group, name, offset, bitmask, callback) {
+    // Jog wheels have 4 byte input
+    message.addControl(group, name, offset, "I", bitmask);
+    message.setCallback(group, name, callback);
+};
+
+TraktorS2MK3.registerInputScaler = function (message, group, name, offset, bitmask, callback) {
+    message.addControl(group, name, offset, "H", bitmask);
+    message.setCallback(group, name, callback);
+};
+
+TraktorS2MK3.registerInputButton = function (message, group, name, offset, bitmask, callback) {
+    message.addControl(group, name, offset, "B", bitmask);
+    message.setCallback(group, name, callback);
+};
+
+TraktorS2MK3.playHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue(field.group, "cue_set", field.value);
+    } else {
+        var playing = engine.getValue(field.group, "play");
+        engine.setValue(field.group, "play", !playing);
+    }
+};
+
+TraktorS2MK3.cueHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue(field.group, "cue_gotoandstop", field.value);
+    } else {
+        engine.setValue(field.group, "cue_default", field.value);
+    }
+};
+
+TraktorS2MK3.shiftHandler = function (field) {
+    TraktorS2MK3.shiftPressed[field.group] = field.value;
+    engine.setValue("[Controls]", "touch_shift", field.value);
+    TraktorS2MK3.outputHandler(field.value, field.group, "shift");
+};
+
+TraktorS2MK3.keylockHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    var keylock = engine.getValue(field.group, "keylock");
+    engine.setValue(field.group, "keylock", !keylock);
+};
+
+TraktorS2MK3.syncHandler = function (field) {
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue(field.group, "beatsync_phase", field.value);
+        // Light LED while pressed
+        TraktorS2MK3.outputHandler(field.value, field.group, "sync_enabled");
+    } else {
+        if (field.value) {
+            if (!TraktorS2MK3.syncPressedTimer[field.group]) {
+                // Start timer to measure how long button is pressed
+                TraktorS2MK3.syncPressedTimer[field.group] = engine.beginTimer(1000, "TraktorS2MK3.syncTimer(\"" + field.group + "\")");
+            }
+
+            var sync = engine.getValue(field.group, "sync_enabled");
+            engine.setValue(field.group, "sync_enabled", !sync);
```

Comment:
I've seen this idiom a few times in your script, but I think you can just do:

```javascript
script.toggleControl(field.group, "sync_enabled")
```

Reply:
The definition of toggleControl is quite close to my idiom:
```javascript
script.toggleControl = function(group, control) {
    engine.setValue(group, control, !(engine.getValue(group, control)))
}
```
Using this function might save a few lines of code, but I dont use any other functions from _common-controller-scripts.js_ and i need to include it in this case (in the XML file). So i would prefer to skip this change request.

Reply:
common-controller-scripts.js is included into every script automatically by Mixxx. It is like a standard library for the scripting environment. Please make use of this function instead of reinventing the wheel.

Reply:
Ok, i changed this in https://github.com/mi01/mixxx/commit/1bdd1e31add3fe11768818c1cb18c4b78b419722